iroh-net/src/magicsock.rs

```diff
@@ -677,6 +1387,15 @@ struct DiscoInfo {
     last_ping_time: Option<Instant>,
 }
 
+#[derive(Debug)]
+enum DerpRecvResult {
+    Error(io::Error),
+    Ok {
+        meta: quinn_udp::RecvMeta,
+        bytes: Bytes,
+    },
```

Comment:
would be nicer to make this an actual `Result`

```rust
type DerpRecvResult = io::Result<(quinn_udp::RecvMeta, Bytes)>;
```

Reply:
did that